File: hu.bme.mit.incqueryd.runtime/hu.bme.mit.incqueryd.agent/src/main/java/hu/bme/mit/incqueryd/agent/Agent.java
package hu.bme.mit.incqueryd.agent;

import io.dropwizard.Application;
import io.dropwizard.Configuration;
import io.dropwizard.setup.Bootstrap;
import io.dropwizard.setup.Environment;

public class Agent extends Application<Configuration> {

	public static void main(String[] args) throws Exception {
		new Agent().run(args);
	}

	@Override
	public void initialize(Bootstrap<Configuration> bootstrap) {
	}

	@Override
	public void run(Configuration configuration, Environment environment) throws Exception {
		DeployResource deployResource = new DeployResource();
		environment.jersey().register(deployResource);
	}

}

File: hu.bme.mit.incqueryd.agent/src/main/scala/hu/bme/mit/incqueryd/agent/Agent.scala
package hu.bme.mit.incqueryd.agent

import io.dropwizard.Application
import io.dropwizard.Configuration
import io.dropwizard.setup.Bootstrap
import io.dropwizard.setup.Environment

object Agent {

  def main(args: Array[String]) {
    new Agent().run(args)
  }

}

class Agent extends Application[Configuration] {

  override def initialize(bootstrap: Bootstrap[Configuration]) {
  }

  override def run(configuration: Configuration, environment: Environment) {
    List(new PrepareInfrastructureResource,
      new DestroyInfrastructureResource,
      new StartMicrokernelsResource).foreach(environment.jersey.register(_))
  }

}
